### PR TITLE
Windows build with VS2017

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
This PR updates the tree such that it can build and run all tests under the latest Visual Studio under Windows.

This PR is dependent on https://github.com/xdrpp/xdrpp/pull/12

notable commits in this PR:
* I had to bump xdrpp, libsodium and easylogging++ to their latest stable releases
* In order to update easylogging++ I had to revert #1272 as it made changes to the easylogging++ header file

### CODE REVIEW

File: src/historywork/ApplyLedgerChainWork.h [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 #pragma once
 
+#include "work/Work.h"
 #include "herder/TxSetFrame.h"
```
Comment (vogel) [inline src/historywork/ApplyLedgerChainWork.h]: this will get reverted by clang-format, maye add an empty line after work/Work.h?

File: src/test/TestPrinter.cpp [hunk 2]
```diff
@@ -6,14 +6,6 @@
 
 namespace Catch
 {
-
-template <>
```
Comment (vogel) [inline src/test/TestPrinter.cpp]: this is useful for checking why tests failed in test outputs - it is not used directly, so removing it does not fail compilation

why is this removed?

File: src/crypto/StrKey.h [hunk 3]
```diff
@@ -14,13 +14,14 @@ struct SecretValue;
 namespace strKey
 {
 
-typedef enum {
```
Comment (vogel) [inline src/crypto/StrKey.h]: I like this!

File: src/historywork/ApplyLedgerChainWork.h (hunk 1 above)
Comment (MonsieurNicolas) [reply src/historywork/ApplyLedgerChainWork.h]: I thought I ran clang-format... but I didn't. Thanks for spotting this.

File: src/test/TestPrinter.cpp (hunk 2 above)
Comment (MonsieurNicolas) [reply src/test/TestPrinter.cpp]: I had to remove it because Visual C++ fails with this code at link time because it says it's a duplicate symbol. I will create a separate issue to look into this with the error output.

Comment (MonsieurNicolas) [reply src/test/TestPrinter.cpp]: nm, found the problem: standard says that template specializations have to be defined before the first instantiation of the template (or else end up with multiple implementations specific to the instantiation sites). By including TestPrinter.h in the right places, it resolved this issue

**Comment by MonsieurNicolas** [comment]
resolves #1312

**Comment by MonsieurNicolas** [comment]
@vogel @jedmccaleb - OK this is ready to go, I am using the latest xdrpp changes and they are good enough to unblock this PR.

I rebased it on top of latest master.

There are very few warnings left - that we'll probably deal with in upcoming PRs (I wanted to deal with the most annoying ones first).

**Comment by jedmccaleb** [comment]
r+ 56def65

**Comment by MonsieurNicolas** [comment]
build picked up old version of libsodium available on build machine, updated configure.ac to pick up matching version that we have vendored in our codebase.

**Comment by MonsieurNicolas** [comment]
r+ 37f1398e7a82c489807c1e518e1b420c42e7835c

**Comment by MonsieurNicolas** [comment]
configure fail, trying again

**Comment by MonsieurNicolas** [comment]
r+ 861fe9fb8cf6a8aeae198243f9df5345ca41106d

**Comment by MonsieurNicolas** [comment]
cleared travis cache - it looks like libsodium is one of the cached libraries

**Comment by MonsieurNicolas** [comment]
r+ 055d1c39d9aa2428041682e0b4209981f506fa3b

**Comment by MonsieurNicolas** [comment]
figured out that it was only that we needed to disable certain files in libsodium in order to compile with asan enabled

**Comment by MonsieurNicolas** [comment]
r+ 1ac571527e1d31cb5e9ed159ff9e85066907ef1e
